git_perf/src/config.rs

```diff
@@ -1,55 +1,153 @@
 use anyhow::Result;
+use config::{Config, ConfigError, File, FileFormat};
 use std::{
-    fs::File,
+    env,
+    fs::{self, File as StdFile},
     io::{Read, Write},
+    path::{Path, PathBuf},
 };
 use toml_edit::{value, Document};
 
-use crate::git::git_interop::get_head_revision;
+use crate::git::git_interop::{get_head_revision, get_repository_root};
 
 // Import the CLI types for dispersion method
 use git_perf_cli_types::DispersionMethod;
 
+/// Get the main repository config path (always in repo root)
+fn get_main_config_path() -> PathBuf {
+    // Use git to find the repository root - hard fail if not found
+    let repo_root = get_repository_root()
+        .expect("Failed to determine repository root - must be run from within a git repository");
+
+    if repo_root.is_empty() {
+        panic!("Repository root is empty - must be run from within a git repository");
+    }
+
+    PathBuf::from(repo_root).join(".gitperfconfig")
+}
+
+/// Write config to the main repository directory (always in repo root)
 pub fn write_config(conf: &str) -> Result<()> {
-    let mut f = File::create(".gitperfconfig")?;
+    let path = get_main_config_path();
+    if let Some(parent) = path.parent() {
+        if !parent.as_os_str().is_empty() {
+            fs::create_dir_all(parent)?;
+        }
+    }
+    let mut f = StdFile::create(path)?;
     f.write_all(conf.as_bytes())?;
     Ok(())
 }
 
-pub fn read_config() -> Result<String> {
-    read_config_from_file(".gitperfconfig")
+/// Read hierarchical configuration (system -> local override)
+pub fn read_hierarchical_config() -> Result<Config, ConfigError> {
+    let mut builder = Config::builder();
+
```

Comment:
@cursoragent this is strange. The repo is always present and the parent directories always exist.